File: salao-backend/src/main/java/com/salao/modules/fiscal/entrada/NotaFiscalEntradaService.java
package com.salao.modules.fiscal.entrada;

import com.salao.modules.compra.Compra;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class NotaFiscalEntradaService {

    private final NotaFiscalEntradaRepository repository;

    public List<NotaFiscalEntradaResponseDTO> listar() {
        return repository.findAll().stream().map(NotaFiscalEntradaResponseDTO::from).toList();
    }

    public NotaFiscalEntradaResponseDTO buscarPorId(Long id) {
        return NotaFiscalEntradaResponseDTO.from(buscarEntidade(id));
    }

    @Transactional
    public NotaFiscalEntrada criarParaCompra(Compra compra) {
        var nota = repository.save(NotaFiscalEntrada.builder()
                .compra(compra)
                .fornecedor(compra.getFornecedor())
                .valorTotal(compra.getValorTotal())
                .serie("1")
                .build());
        nota.setNumeroNota("NFE-" + String.format("%06d", nota.getId()));
        return repository.save(nota);
    }

    public NotaFiscalEntradaResponseDTO atualizarDadosTransporte(Long id, String transportadoraNome, String veiculoPlaca) {
        var nota = buscarEntidade(id);
        nota.setTransportadoraNome(transportadoraNome);
        nota.setVeiculoPlaca(veiculoPlaca);
        return NotaFiscalEntradaResponseDTO.from(repository.save(nota));
    }

    private NotaFiscalEntrada buscarEntidade(Long id) {
        return repository.findById(id)
                .orElseThrow(() -> new RuntimeException("Nota fiscal de entrada não encontrada"));
    }
}

File: salao-frontend/src/services/notaFiscalEntradaService.ts
import axios from 'axios';

const API = 'http://localhost:8080/api';

export interface NotaFiscalEntrada {
  id: number;
  numeroNota: string;
  serie?: string;
  dataEmissao: string;
  fornecedor?: { id: number; fornecedor: string };
  valorTotal: number;
  transportadoraNome?: string;
  veiculoPlaca?: string;
  compraId?: number;
}

export interface TransporteRequest {
  transportadoraNome?: string;
  veiculoPlaca?: string;
}

export const notaFiscalEntradaService = {
  listar:              ()                                      => axios.get<NotaFiscalEntrada[]>(`${API}/notas-fiscais-entrada`),
  buscarPorId:         (id: number)                           => axios.get<NotaFiscalEntrada>(`${API}/notas-fiscais-entrada/${id}`),
  atualizarTransporte: (id: number, data: TransporteRequest)  => axios.put(`${API}/notas-fiscais-entrada/${id}/transporte`, data),
};
